kore/src/Kore/Step/Step.hs

```diff
@@ -135,6 +137,12 @@ deriving instance Ord (variable Object) => Ord (Result variable)
 
 deriving instance Show (variable Object) => Show (Result variable)
 
+instance TopBottom (Result variable) where
+    isTop (Result {unifiedRule, result}) =
+        isTop (Predicated.withoutTerm unifiedRule) && isTop result
```

Comment:
The `PredicateSubstitution` from `unifiedRule` is always included in `result`, so we only need to check the `result`.

Reply:
Yes, but there is nothing forcing them to be in sync, so I guess that checking both is better.

Reply:
Then we should throw an `error` if they are not in sync, because if they are not kept in sync then the result will be wrong.